jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -378,14 +473,25 @@ public String toString()
          * detected by the {@link ManagedSelector} for this endpoint.
          *
          * @return a job that may block or null
+         * @param key the selected SelectionKey
          */
-        Runnable onSelected();
+        Runnable onSelected(SelectionKey key);
 
         /**
          * Callback method invoked when all the keys selected by the
          * {@link ManagedSelector} for this endpoint have been processed.
+         * @param key the SelectionKey to update
+         */
+        void updateKey(SelectionKey key);
+
+        /**
+         * Callback method invoked when a selectable is transferred
+         * from one selector to a new selector.
+         *
+         * @param oldKey the old SelectionKey
+         * @param newKey the new SelectionKey
          */
-        void updateKey();
+        void replaceKey(SelectionKey oldKey, SelectionKey newKey);
```

Comment:
Both implementations of this method don't use the `oldKey` parameter. Use it or drop it?

Reply:
I think it is strange to have both replaceKey and to pass in the key to onSelected etc.
If you are passing in the key for the main usages, then findKey is probably OK for the remaining cases.
So get rid of one mechanism or the other.